src/libsyntax/parse/token.rs

```diff
@@ -201,6 +201,7 @@ impl Token {
             OpenDelim(..)               => true, // tuple, array or block
             Literal(..)                 => true, // literal
             Not                         => true, // operator not
+            Tilde                       => true, // operator not (incorrect)
```

Comment:
This is slightly risky, `can_begin_expr` can affect syntax accepted by parser in general, not only diagnostics.
`~` is not used anywhere right now, so it's ok for now, but it may be used in the future.